src/main/java/com/example/account/elastic/repository/impl/AccountCriteriaQueryImpl.java

```diff
@@ -0,0 +1,48 @@
+package com.example.account.elastic.repository.impl;
+
+import com.example.account.elastic.data.criteria.AccountCriteria;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.elasticsearch.core.query.Criteria;
+import org.springframework.data.elasticsearch.core.query.CriteriaQuery;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+public class AccountCriteriaQueryImpl implements AccountCriteriaQuery {
+
+    @Override
+    public CriteriaQuery createAccountCriteriaQuery(AccountCriteria accountCriteria) {
+        log.debug("Account criteria entity " + accountCriteria);
+        Criteria criteria = new Criteria();
+        if (accountCriteria.getName() != null) {
```

Comment:
@amerkul try to refactor this logic to the object manner
simply because the current version of this method is not scalable
if you want to add/remove some criteria fields you will need to change this code

each field like `name`, `balance`, etc. can be converted into objects
then you can apply all of them to criteria in declarative way